cloudinit/log.py

```diff
@@ -27,12 +27,6 @@
 def setup_basic_logging(level=logging.DEBUG, formatter=None):
     formatter = formatter or logging.Formatter(DEFAULT_LOG_FORMAT)
     root = logging.getLogger()
-    for handler in root.handlers:
```

Comment:
Good drop, I think this was originally intended to handle re-entrant calls to setup_basic_logging where stderr stream was already setup to avoid duplication of logs to stderr. I'll double check that this is no longer an issue due to how you've changed logging call-sites in your previous logging refactors.

Reply:
This looks like it'll mean that if we get multiple calls to `setup_basic_logging(<log_level>)` then we'll have duplicated logs to stderr for each LOG.root.handler managinghandler. In practice though, the CLI commands we are only calling this once in any case during handling of CLI arguments, so adding the streamhandler for stderr at a given log level will properly configure the default  log-level here and I don't think we really need to solve issues with this function not being idempotent.